webapp/src/ts/modals/edit-user/update-password.component.ts

```diff
@@ -48,13 +53,27 @@ export class UpdatePasswordComponent extends EditUserAbstract implements OnInit
     this.errors = {};
     this.setProcessing();
     if (this.validatePasswordFields()) {
-      const updates = { password: this.editUserModel.password };
+      const password = this.editUserModel.password;
+      const updates = { password };
       const username = this.editUserModel.username;
       this.updateUserService
         .update(username, updates, username, this.editUserModel.currentPassword)
         .then(() => {
-          this.setFinished();
-          this.windowReload();
+          const data = this.getLoginData(username, password);
+          this.userLoginService
```

Comment:
Returning this promise here, along with returning the wrapping promise `this.updateUserService` will allow you to `await` this action in the unit test, and you won't need `done` or `setTimeout`.

Reply:
Thank you for the suggestion.